extensions/script-libraries/mime/mime.livecodescript

```diff
@@ -0,0 +1,1558 @@
+
+*/
+
+command MIMEEncode pData, pEncoding
+   local tData
+   
+   put pData into tData
+   if tData is not strictly a binary string then
+      put textEncode(tData, "UTF-8") into tData
+   end if
+   
+   switch pEncoding
+      case "quoted-printable"
+         __EncodeQuotedPrintable tData
+         break
+      case "base64"
+         __EncodeBase64 tData
+         break
+      default
+         return "unsupported encoding" for error
+   end switch
+   
+   return tData for value
+end MIMEEncode
+
+private command __EncodeBase64 @xData
+   put replaceText(base64Encode(xData), return, crlf) into xData
+end __EncodeBase64
+
+private command __EncodeQuotedPrintable @xData
+   local tData
+   local tLineLength = 1
+   local tNumBytes
+   
+   put the number of bytes of xData into tNumBytes
+   
+   local tIndex = 1
+   repeat for each byte tByte in xData
+      if tLineLength is 76 then
+         put "=" & crlf after tData
+         put 1 into tLineLength
+      end if
+      
+      local tByteNum
+      put byteToNum(tByte) into tByteNum
+      
+      local tPrevByteNum
+      
+      switch
+         case tByteNum is 10 and tPrevByteNum is not 13
+            put crlf after tData
+            put 1 into tLineLength
+            break 
+         case (tByteNum is 9 or tByteNum is 32) and  tLineLength is not 76 and tIndex is not tNumBytes
+         case tByteNum >= 33 and tByteNum <= 126 and tByteNum is not 61
+            put tByte after tData
+            add 1 to tLineLength
+            break
+         default
+            if tLineLength >= 74 and \
+                  not (tLineLength is 74 and tIndex is tNumBytes) then
+               put "=" & crlf after tData
+               put 3 into tLineLength
+            end if
+            
+            local tConverted
+            put baseConvert(tByteNum,10,16) into tConverted
+            if the length of tConverted is 1 then
+               put 0 before tConverted
+            end if
+            put "=" & tConverted after tData
+            add 3 to tLineLength
+            break
+      end switch
+      
+      put tByteNum into tPrevByteNum
+      add 1 to tIndex
+   end repeat
+   
+   put tData into xData
+end __EncodeQuotedPrintable
+
+/**
+
+Build a multipart mime content type from an array of pre-encoded content
+
+Example:
+put "text/html" into tParamA["type"]
+MIMECreateMultipartContent tContentA, "related", tParamA
+put it into tMailBody
+
+Description:
+This command takes an array of pre-encoded content and creates a multipart
+content type header and boundary to build a single multipart encoded value.
+
+Parameters:
+pContentA (array):
+A numerically indexed array of pre-encoded parts including headers. If no headers are present
+a leading crlf should be present.
+
+pMultipartType (optional string):
+The type of multipart content. If not provided the default type will be `mixed`.
+
+pParamA (optional array):
+Additional parameter array in key : value pairs
```

Comment:
This could maybe do with a little clarification about what is meant by "additional parameters" - upon first reading, I interpreted it as indicating they were additional settings controlling the behaviour of the function rather than key=value options added to the Content-Type header.